src/dstore/pmix_esh.c

```diff
@@ -701,29 +1024,42 @@ static int _pmix_getpagesize(void)
 #endif
 }
 
-static seg_desc_t *_create_new_segment(segment_type type, char *nsname, uint32_t id)
+static seg_desc_t *_create_new_segment(segment_type type, const char *nsname, uint32_t id)
 {
     int rc;
     char file_name[PMIX_PATH_MAX];
     size_t size;
     seg_desc_t *new_seg = NULL;
+    char *nspace_path = NULL;
+    uid_t jobuid;
 
     PMIX_OUTPUT_VERBOSE((10, pmix_globals.debug_output,
                          "%s:%d:%s: segment type %d, nspace %s, id %u",
                          __FILE__, __LINE__, __func__, type, nsname, id));
 
+    nspace_path =  _esh_nspace_path(nsname);
+    if (NULL == nspace_path) {
+        PMIX_ERROR_LOG(PMIX_ERROR);
+        return NULL;
+    }
+
+    if (PMIX_SUCCESS != _esh_nspace_jobuid(nsname, &jobuid)) {
+        PMIX_ERROR_LOG(PMIX_ERROR);
+        return NULL;
+    }
+
     switch (type) {
         case INITIAL_SEGMENT:
             size = _initial_segment_size;
-            snprintf(file_name, PMIX_PATH_MAX, "%s/%s_initial-pmix_shared-segment-%u", _nspace_path, _unique_id(), id);
+            snprintf(file_name, PMIX_PATH_MAX, "%s/%u_initial-pmix_shared-segment-%u", nspace_path, jobuid, id);
```

Comment:
why do we need the `jobuid` in the filename? All those files should be located under
<tmpdir>/<jobuid> directory so no need to use `jobuid` in the filenames!